hack/check/imagelinter/main.go

```diff
@@ -54,168 +54,197 @@ func init() {
 	}
-	return false, err
+	return nil
 }
 
-func InitialChecks(imc *imglint.ImageLintConfig, wrapper *imgwrapper.Wrapper, key string) (cont bool, err error) {
-	_, err = wrapper.PullImage()
+// CheckLicense checks if the container declares an incompatible license. If so it returns an error, otherwise
+// if the license is fine or if it could not be determined it will return nil.
+func CheckLicense(wrapper *imgwrapper.Wrapper, key string) error {
+	localName := fmt.Sprintf("./LICENSE-%s", key)
+	_, err := wrapper.ContainerCP("/licenses/LICENSE", localName)
 	if err != nil {
-		imc.OnEvent("Pull Failed", "Pulling Image Failed:"+err.Error(), key)
-		return true, err
+		// No license file to check
+		return nil
 	}
-	skip, err := wrapper.Validate(imc.SuccessValidators)
-	if skip && err == nil {
-		imc.OnEvent("Pass", "According to the image history, this is not an Alpine Image", key)
-		return true, nil
+
+	license, err := os.ReadFile(localName)
+	defer os.Remove(localName)
+	if err != nil {
+		// Unable to read the LICENSE file, punt
+		return nil
 	}
-	return false, nil
-}
 
-func LintAll(imc *imglint.ImageLintConfig, wrapper *imgwrapper.Wrapper, key string) (bool, error) {
-	cont, err := InitialChecks(imc, wrapper, key)
-	if cont {
-		return true, err
+	// TODO: Add checks for other known incompatible licenses
+	if strings.Contains(string(license), "Alpine") {
+		return errors.New("alpine license")
 	}
-	_, err = wrapper.CreateContainer()
+	return nil
+}
+
+// IsBusyBox checks if the container image is using busybox. If so, returns an
+// error, else if it is not or can't be determined returns nil.
+func IsBusyBox(wrapper *imgwrapper.Wrapper) error { //nolint:unparam
+	_, err := wrapper.RunCommand("exec", wrapper.Container, "/bin/busybox")
 	if err != nil {
-		return false, err
-	}
-	if wrapper.IsContainerExists() {
-		_, err := wrapper.ContainerCP("/etc/os-release", "./")
-		if err == nil {
-			fatal, cont, err := IsAlpine(imc, wrapper, key)
-			if fatal {
-				isFatal = fatal
-			}
-			if cont {
-				return true, err
-			}
-		}
-		_, err = wrapper.ContainerCP("/usr/lib/os-release", "./")
-		if err == nil {
-			fatal, cont, err := IsAlpine(imc, wrapper, key)
-			if fatal {
-				isFatal = fatal
-			}
-			if cont {
-				return true, err
-			}
-		}
-		_, err = wrapper.ContainerCP("/licenses/LICENSE", "./")
-		if err == nil {
-			cont, err := HasLicense(imc, wrapper, key)
-			if cont {
-				return true, err
-			}
-		}
-		_, err = wrapper.RunCommand("run", `--entrypoint=/bin/busybox`, "--name", wrapper.Container+"a", key)
-		if err == nil {
-			cont, err := IsBusyBox(imc, wrapper, key)
-			if cont {
-				return true, err
-			}
-		}
+		// Successfully ran busybox
+		return nil
 	}
-	return false, nil
+
+	// TODO: What the heck are we trying to do here? The original code was checking
+	// if it was busybox, and if so just reporting success that it wasn't alpine.
+	// Do we care about busybox? What does that have to do with alpine?
```

Comment:
At one point there were a number of alpine based images included in tce (pre - GA) and there was this effort to remove them: https://github.com/vmware-tanzu/community-edition/issues/1553

I _believe_ the imagelinter was intended to make sure we weren't shipping alpine images but it was _really_ brittle and didn't actually inspect the kbld / ytt output of images to see if _those_ images were being built using alpine: https://github.com/vmware-tanzu/community-edition/issues/1723

_I think we should throw this code away and start over or drastically refactor._

For further reference:
- https://github.com/vmware-tanzu/tanzu-framework/issues/210
- https://github.com/vmware-tanzu/community-edition/issues/685
- https://github.com/vmware-tanzu/community-edition/issues/1625
- https://github.com/vmware-tanzu/community-edition/issues/1173#issuecomment-886557734
- https://github.com/vmware-tanzu/community-edition/pull/1302

Reply:
Yes, that was why this check was added.

Just to make sure the context is set right, the code comment you commented on was related to checking for busybox, not a larger existential question for the whole tool.

Reply:
Hmmm yeah I'm not sure either. I don't know why `busybox` was checked against explicitly.